demo/api/src/warnings/warning-checker.console.ts

```diff
@@ -0,0 +1,83 @@
+import { FlatBlocks } from "@comet/blocks-api";
+import { DiscoverService } from "@comet/cms-api/lib/dependencies/discover.service";
+import { Connection, CreateRequestContext, MikroORM } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { SeoBlock } from "@src/pages/blocks/seo.block";
+import hasha from "hasha";
+import { Command, Console } from "nestjs-console";
+
+import { Warning } from "./entities/warning.entity";
+import { WarningSeverity } from "./entities/warning-severity.enum";
+
+@Injectable()
+@Console()
+export class WarningCheckerConsole {
+    private connection: Connection;
+    constructor(
+        private readonly orm: MikroORM,
+        private readonly discoverService: DiscoverService,
+        private readonly entityManager: EntityManager,
+        @InjectRepository(Warning) private readonly warningsRepository: EntityRepository<Warning>,
+    ) {
+        this.connection = entityManager.getConnection();
+    }
+
+    @Command({
+        command: "check-warnings",
+        description: "Checks for warnings",
+    })
+    @CreateRequestContext()
+    async execute(): Promise<void> {
+        // TODO: Check if data itself is valid in the database. (Maybe some data was put into database and is not correct or a migration was done wrong)
+
+        for (const rootBlockEntity of this.discoverService.discoverRootBlocks()) {
+            const { metadata, column, block } = rootBlockEntity;
+            const sql = `SELECT id, "${column}" as "block" FROM "${metadata.tableName}";`;
+            const rootBlocks = await this.connection.execute(sql);
+
+            for (const rootBlock of rootBlocks) {
+                const blockData = SeoBlock.blockInputFactory(rootBlock.block).transformToBlockData(); // don't use SeoBlock but maybe a generic Block. I need to call transformToBlockData o therwise the FlatBlocks function does not work
+                // const blockData = block.blockDataFactory(rootBlock.block); // maybe something like this is the correct way - but this here does not work
+
+                const flatBlocks = new FlatBlocks(blockData, {
+                    name: block.name,
+                    visible: true,
+                    rootPath: "root",
+                });
+                for (const node of flatBlocks.depthFirst()) {
+                    const warnings = node.block.reportWarnings();
+
+                    if (warnings.length > 0) {
+                        // TODO: auto resolve warnings
+
+                        for (const warning of warnings) {
+                            const type = "Block";
+                            const uniqueIdentifier = `${type};${node.pathToString()};${warning.message}`; // TODO: Is this enough to be unique?
+                            const hash = hasha(uniqueIdentifier, { algorithm: "md5" });
```

Comment:
I'm not sure if this should be hashed. We'll probably need the path to navigate to the warning in the Admin.

Reply:
yes makes no sense (yet) and maybe not at all as you said:
5cab6cb11bc02ebd3a7ba7710600301968d7fcac

Reply:
I thought about it again, and I need 2 things:

 - an id with which i can check if the warning already exists for autoresolving
 - metadata, so I can navigate in the admin to the warning. Also for custom warnings in the future, when the projects adds for example news. I'm thinking of a json field in the entity. I'd add this in future PRs

I also think generating the id from the block id + message should be enough for this case. The id from the block is a uuid which should be unique enough.

I removed the uniqueIdentifier in this commit: 9a76644e3507b51ec076eed5e5e52f096a93635c
I'm now using the default id field and generating the id from a string.

What do you think about that?